SWLOR.Game.Server/Item/RepairKit.cs

```diff
@@ -31,20 +33,47 @@ public CustomData StartUseItem(NWCreature user, NWItem item, NWObject target, Lo
         public void ApplyEffects(NWCreature user, NWItem item, NWObject target, Location targetLocation, CustomData customData)
         {
             SkillType skillType = GetSkillType(item);
+            NWItem targetitem = (target.Object);
             int tech = item.GetLocalInt("TECH_LEVEL");
             float maxDurabilityReductionPenalty = item.GetLocalFloat("MAX_DURABILITY_REDUCTION_PENALTY");
             int repairAmount = tech * 2;
-
+            int skillrank;
+            int Level = targetitem.RecommendedLevel;
+            int delta = 0;
+            int baseXP = 0;
             if (skillType == SkillType.Armorsmith)
-                repairAmount += item.CraftBonusArmorsmith + (_perk.GetPCPerkLevel(user.Object,PerkType.ArmorRepair) * 2);
+            {
+                skillrank = (_skill.GetPCSkillRank((NWPlayer)user, skillType));
+                repairAmount += item.CraftBonusArmorsmith + (_perk.GetPCPerkLevel(user.Object, PerkType.ArmorRepair) * 2);
+                delta = Level - skillrank;
+            }
             else if (skillType == SkillType.Weaponsmith)
-                repairAmount += item.CraftBonusWeaponsmith + (_perk.GetPCPerkLevel(user.Object,PerkType.WeaponRepair) * 2);
+            {
+                skillrank = (_skill.GetPCSkillRank((NWPlayer)user, skillType));
+                repairAmount += item.CraftBonusWeaponsmith + (_perk.GetPCPerkLevel(user.Object, PerkType.WeaponRepair) * 2);
+                delta = Level - skillrank;
+            }
             else if (skillType == SkillType.Engineering)
-                repairAmount += item.CraftBonusEngineering + (_perk.GetPCPerkLevel(user.Object,PerkType.ElectronicRepair) * 2);
+            {
+                skillrank = (_skill.GetPCSkillRank((NWPlayer)user, skillType));
+                repairAmount += item.CraftBonusEngineering + (_perk.GetPCPerkLevel(user.Object, PerkType.ElectronicRepair) * 2);
+                delta = Level - skillrank;
+            }
             float minReduction = 0.05f * tech;
             float maxReduction = 0.15f * tech;
             float reductionAmount = _random.RandomFloat(minReduction, maxReduction);
-
+            if (delta >= 6) baseXP = 400;
+            else if (delta == 5) baseXP = 350;
+            else if (delta == 4) baseXP = 325;
+            else if (delta == 3) baseXP = 300;
+            else if (delta == 2) baseXP = 250;
+            else if (delta == 1) baseXP = 225;
+            else if (delta == 0) baseXP = 200;
+            else if (delta == -1) baseXP = 150;
+            else if (delta == -2) baseXP = 100;
+            else if (delta == -3) baseXP = 50;
+            else if (delta == -4) baseXP = 25;
+            _skill.GiveSkillXP((NWPlayer)user,skillType,baseXP,true);
```

Comment:
We've reused this case statement a bunch of times. It might make sense for us to put it in a central location, like the SkillService. You don't need to do anything with this, I'm just kind of reminding myself. It's fine how it is at the moment.